From cc2e1f8ee1d28a772c0309b226e996359180124a Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sun, 16 Sep 2018 16:35:41 +0000
Subject: [PATCH 15/22] test/linkage-visibility: Ensure symbols are visible to
 dlsym

DynamicLibrary uses libc's dlsym() function internally to find symbols.
Some implementations of dlsym(), like musl's, only look at dynamically-
exported symbols, as found in shared libraries. To also export symbols
from the main executable, pass --export-dynamic to the linker.
---
 src/test/run-pass-fulldeps/auxiliary/linkage-visibility.rs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/test/run-pass-fulldeps/auxiliary/linkage-visibility.rs b/src/test/run-pass-fulldeps/auxiliary/linkage-visibility.rs
index 4ea3d0d0d0..b958e1a1fc 100644
--- a/src/test/run-pass-fulldeps/auxiliary/linkage-visibility.rs
+++ b/src/test/run-pass-fulldeps/auxiliary/linkage-visibility.rs
@@ -8,6 +8,8 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+// compile-flags: -C link-arg=-Wl,--export-dynamic
+
 #![feature(rustc_private)]
 
 // We're testing linkage visibility; the compiler warns us, but we want to
-- 
2.19.2

